compiler/noirc_evaluator/src/ssa/ir/instruction/call.rs

```diff
@@ -60,9 +60,14 @@ pub(super) fn simplify_call(
                 } else {
                     unreachable!("ICE: Intrinsic::ToRadix return type must be array")
                 };
-                let result_array = constant_to_radix(endian, field, 2, limb_count, dfg);
-
-                SimplifyResult::SimplifiedTo(result_array)
+                if limb_count < field.num_bits() {
```

Comment:
I would move this logic inside `constant_to_radix`, where we can check if limb_count is at least decomposed_integer.len() and return an optional valueId

Reply:
Yeah, that's a good point. I've made this change.